tests/importer/importer.cpp

```diff
@@ -56,10 +56,9 @@ TEST(Importer, warningCircularImportReferencesComponent)
 {
     const std::string errorMessage =
         "Cyclic dependencies were found when attempting to resolve a component in the model 'circularImport1'. The dependency loop is:\n"
-        " - component 'i_am_cyclic' is imported from 'c2' in 'circularImport_2.cellml';\n"
-        " - component 'c2' is imported from 'c3' in 'circularImport_3.cellml';\n"
-        " - component 'c3' is imported from 'i_am_cyclic' in 'circularImport_1.cellml'; and\n"
-        " - component 'i_am_cyclic' is imported from 'c2' in 'circularImport_2.cellml'.";
+        " - the component 'i_am_cyclic' is referencing a component 'c2' in the model 'circularImport2' imported from 'circularImport_2.cellml';\n"
+        " - the component 'c2' is referencing a component 'c3' in the model 'circularImport3' imported from 'circularImport_3.cellml'; and\n"
+        " - the component 'c3' is referencing a component 'i_am_cyclic' in the model 'circularImport1' imported from 'circularImport_1.cellml'.";
```

Comment:
this set of messages is not defining a circular error as there is nothing to say that `c3` is importing the same `i_am_cyclic` component...

Reply:
?? We are saying that "Cyclic dependencies were found when attempting to resolve a component in the model 'circularImport1'" and that `c3` references "'i_am_cyclic' in the model 'circularImport1' imported from 'circularImport_1.cellml'". So...?

That aside, I would reword the error message, which I find too verbose:
```
Cyclic dependencies were found when attempting to resolve component 'i_am_cyclic' in 'circularImport_1.cellml'. The dependency loop is:
 - component 'i_am_cyclic' references component 'c2' in 'circularImport_2.cellml';
 - component 'c2' references component 'c3' in 'circularImport_3.cellml';
 - component 'c3' references component 'i_am_cyclic' in 'circularImport_1.cellml'.
```

(There is only one model in a CellML file, so no need to mention it. To mention the CellML filename is sufficient.)

Reply:
We seem to have one vote each way here.

I see `I_am_cyclic -> c2 -> c3 -> i_am_cyclic` which shows a loop.  Do you disagree with that @nickerso?

Reply:
LOL - I can't read. Didn't see the first line in the error message as it was not highlighted by Github as a difference - so thanks to @agarny in the above comment making that clearer.

I'd prefer to see the href's resolved to absolute URLs as that is the information I actually need to debug circular imports as the filenames on their own are pretty meaningless, but that can be a future fix if needed.

Reply:
> I'd prefer to see the href's resolved to absolute URLs as that is the information I actually need to debug circular imports as the filenames on their own are pretty meaningless, but that can be a future fix if needed.

Agreed with absolute URLs. One might, for some reasons, have one or several CellML files with the same name.